commands.py

```diff
@@ -26,18 +58,25 @@ def run(self, edit: sublime.Edit) -> None:
             return
 
         region = completion.region
-        display_text = completion.display_text or ""
-
         if not region:
             return
 
         completion.hide()
-        self.view.insert(edit, region[1], display_text)
+        self.view.insert(edit, region[1], completion.display_text)
```

Comment:
This might not work, as we reset `display_text` when we hide, so it would be an empty string here.  
We should keep `display_text = completion.display_text or ""` for the accept command

Reply:
Wouldn't the case you say be guarded by 
```py
        if not region:
            return
```
? If not, I think we should reset region when hide.

Reply:
@timfjord So as per @jfcherng should we just call insert first and then hide? This would remove the empty string issue. We could also delete the key vs erase the key.

Reply:
Why we have to reset something when doing `hide()`?

Reply:
Well, I think the better function name is `clear` which would clear the pop up, clear associated completion metadata. 

Either way, I think that is more of a refactor with the Completion class. I'll fix this PR and merge, and then we need to refactor Completion

Reply:
We reset the variables because we rely on theme in [the context calculation ](https://github.com/TheSecEng/LSP-copilot/blob/master/listeners.py#L34-L37)

There is `Completion.is_visible` that is responsible for that and it is checking whether both the `region` and the `dispaly_text` are present.

I think these two concerns(the UI part and the metadata) are related, and that's why they are grouped together.  
When we hide the popup we probably don't want the metadata to stay

Reply:
> We reset the variables because we rely on them in [the context calculation ](https://github.com/TheSecEng/LSP-copilot/blob/master/listeners.py?rgh-link-date=2022-07-15T18%3A59%3A05Z#L34-L37)
> 
> There is `Completion.is_visible` that is responsible for that and it is checking whether both the `region` and the `dispaly_text` are present.

That still relies on other properties. I mean something like

```py
    def is_visible(self) -> bool:
        return bool(self._settings("is_visible") or False)
```

Reply:
Changes are proposed on https://github.com/TheSecEng/LSP-copilot/pull/8

Reply:
We could do that, but I think we still need to keep `is_visible` in sync with `display_text`, `region` and `uuid`. And if we hide the popup we should also clean up those variables

Reply:
I see no reason to clean them up at this moment. Except for gaining tiny memory gc.

Reply:
🙂 Yeah, I was basically going to say, that maybe it's not a bad idea to keep them out of sync. This way, theoretically, we could even restore the last suggestion